internal/tofu/eval_context_builtin.go

```diff
@@ -309,12 +310,17 @@ func (ctx *BuiltinEvalContext) EvaluateBlock(body hcl.Body, schema *configschema
 	diags = diags.Append(evalDiags)
 	val, evalDiags := scope.EvalBlock(body, schema)
 	diags = diags.Append(evalDiags)
+	val, depDiags := marks.DeprecatedDiagnosticsInBody(val, body)
```

Comment:
Can we move this call to scope.EvalBlock?

Reply:
Yep, we would also need to put a flag on what sources to ignore so scope will be extended slightly. Changed in [46fc042](https://github.com/opentofu/opentofu/pull/2633/commits/46fc042f1b327b00d1eb0715cdedae7160ab8e01)